src/FreeDSx/Snmp/Module/SecurityModel/UserSecurityModelModule.php

```diff
@@ -122,14 +123,20 @@ public function handleIncomingMessage(AbstractMessageV3 $message, array $options
         if (!$securityParams) {
             throw new SecurityModelException('The received SNMP message is missing the security parameters.');
         }
-        if ($options['use_auth'] && !$header->hasAuthentication()) {
+
+        $useAuth = $options['use_auth'];
+        $usePriv = $options['use_priv'];
+        assert(is_bool($useAuth));
+        assert(is_bool($usePriv));
```

Comment:
I'm not too crazy about the use of assert here. I guess I'd rather throw a SecurityModelException or catch this when the options are being set in the client and throw something there.

Reply:
That is because PHPStan doesn't know what type it is as options are currently array. 

I'd do it later but if we make options as nicely typed object PHPStan will then require to drop asserts here as it will know they're useless (_asserting something is bool while it's already bool_ )

Reply:
Yeah, as I look back I feel that I probably should've used an object for the options. Would've made passing it around far easier as well.

Reply:
So is it fine to leave them there for now? Would have to add an exclusion to phpstan but that is best to be avoided if not really necessary

Reply:
Would doing something like this work with phpstan?
```php
if (!(is_bool($useAuth) && is_bool($usePriv))) {
    throw new InvalidArgumentException('The options use_auth and use_priv must be boolean.');
}
```
I'd still lean towards throwing an actual exception in these cases instead of relying on whether assertions are even enabled or not. Though if it isn't possible, maybe not a huge deal.

Reply:
That's true. On larger projects I use https://github.com/beberlei/assert but IMO it's not suitable for such small library to require another dependency for it